objective.py

```diff
@@ -60,21 +111,57 @@ def compute(
             Callable[[Tensor, int], Tensor],
         ],
     ):
+        """Compute the metrics.
+
+        Input parameters are the output of `Solver.get_result`.
+
+        Parameters
+        ----------
+        result : Tuple[ Callable[[Tensor, Tensor], Tensor], Callable[[Tensor, int], Tensor], ]
+            - log_prob: computes the log probabilities of the approximate posterior
+                for a given batch of observations and parameters.
+                Returns a tensor of shape (batch_size,).
+            - sample: samples from the approximate posterior for a given observation
+                and sample size. Returns a tensor `theta_est` of shape
+
+        Returns
+        -------
+        Dict
+            dictionary of computed metrics
+                - value: metric used as stopping criterion (NLL on test data)
+                - any other metric computed ...
+
+        """  # noqa:E501
+        # Get result: `log_prob`` and `sample` functions from the solver.
         log_prob, sample = result
+
+        # Compute NLL on train and test data.
+        # Always computed. Do not require samples from the reference posterior.
         nll_test = negative_log_lik(log_prob, self.theta_test, self.x_test)
         nll_train = negative_log_lik(log_prob, self.theta_train, self.x_train)
 
+        # Compute metrics on reference data if available.
         if self.theta_ref is None:
             c2st_mean, c2st_std = None, None
             emd_mean, emd_std = None, None
             mmd_mean, mmd_std = None, None
             sampling_time = None
         else:
+            # Sampling from the approximate posterior.
             start = time.perf_counter()
+            # same sample size as for the reference posterior
             n_ref = (theta.shape[0] for theta in self.theta_ref)
+            # same conditioning observation as for the reference posterior
             theta_est = [sample(x, n) for x, n in zip(self.x_ref, n_ref)]
             end = time.perf_counter()
 
+            # Normalize samples.
+            theta_est = [
+                (theta - self.mean_theta) / self.std_theta
+                for theta in theta_est
+            ]
```

Comment:
This is incorrect. The estimators have been trained on normalized data, so they should generate normalized data already.

However we probably have an issue with the sampling from NRE, which uses a prior that does not match the normalized parameters.

Reply:
Oh okay yes indeed my tests fail.. And yes please! 
Do we keep the z-scoring in the c2st metric?

Reply:
I submitted a PR (#12) to fix the issue. I did some tests, and it seems to work on my machine. For the C2ST, it is not necessary to re-scale the data if they are already standardized.